crates/common/src/config.rs

```diff
@@ -113,13 +113,57 @@ const fn default_u256() -> U256 {
 }
 
 #[derive(Debug, Deserialize, Serialize)]
+pub enum ModuleSource {
+    #[serde(rename = "path")]
+    Path(String),
+    #[serde(rename = "docker_image")]
+    DockerImageId(String),
+}
+
+#[derive(Debug, Serialize)]
+#[serde(deny_unknown_fields)]
 pub struct ModuleConfig<T = ()> {
     pub id: String,
-    pub path: String,
+    #[serde(flatten)]
+    pub source: ModuleSource,
     #[serde(flatten)]
     pub extra: T,
 }
 
+impl<'de, T> Deserialize<'de> for ModuleConfig<T>
```

Comment:
That's the quick approach I took to scratch my itch for modelling a strict config structure: exactly one of `path` or `docker_image` must be specified in the `config.example.toml` for a given module - marking the module as either launchable from a raw binary (like it used to be) OR from a docker image (the new addition).

We could choose another approach for modelling the config (i.e. an additional tag specifying the variant, and then the necessary source data (path or docker image id); or similar...)